frontend/src/components/ManageUsers/ManageUsers.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { ReactEventHandler, useEffect, useState } from 'react';
+import { FormControlLabel, Checkbox, Container, Paper, CssBaseline } from '@material-ui/core';
+import styles from './ManageUsers.module.css';
+import AddButton from '../AddButton';
+import SelectSortBy from '../SelectSortBy';
+import SearchInput from '../SearchInput';
+import Table from '../ReusableTable';
+import { filterData, sortData } from '../ReusableTable/ReusableTableSlice';
+import { useAppDispatch } from '../../app/hooks';
+
+
+interface CheckboxProps {
+  name: string;
+  checked: boolean;
+  onChange: ReactEventHandler;
+}
+
+const PrimaryCheckBox: React.FC<CheckboxProps> = ({ name, checked, onChange }) => (
+    <FormControlLabel
+    className={styles.checkbox}
+    control={
+      <Checkbox
+        name={name}
+        checked={checked}
+        onChange={onChange}
+        style={{color: 'white'}}
+      />
+    }
+    label={name}
+  />
+);
+
+export interface ManageUsersProps {
+  getUsers: () => Promise<any[]>;
+  onClickAdd: () => void;
+}
+
+const ManageUsers: React.FC< ManageUsersProps > = ({ getUsers, onClickAdd }) => {
+  const dispatch = useAppDispatch();
+  const [statusFilters, setStatusFilters] = useState({
+    Active: false,
+    Archived: false,
+    Resigned: false,
+  });
+  const [typeFilters, setTypeFilters] = useState({
+    Candidate: false,
+    Participant: false,
+    Mentor: false,
+    Admin: false,
+  });
+  const [sortBy, setSortBy] = useState('');
+  const [search, setSearch] = useState('');
+
+  const changeStatusFilter = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setStatusFilters({ ...statusFilters, [event.target.name]: event.target.checked });
+  };
+  const changeTypeFilter = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTypeFilters({ ...typeFilters, [event.target.name]: event.target.checked });
+  };
+  const changeSortBy = (value: string) => {
+    setSortBy(value);
+  };
+  const changeSearch = (value: string) => {
+    setSearch(value);
+  }
+  
+  useEffect(() => {
+    const typeValues = Object.entries(typeFilters)
+      .filter(([, value]) => value)
+      .map(([key, value]) => key);
+    const statusValues = Object.entries(statusFilters)
+      .filter(([, value]) => value)
+      .map(([key, value]) => key);
+    const filters = [
+      {column: 'type', values: typeValues},
+      {column: 'status', values: statusValues}
+    ];
+    dispatch(filterData({table: 'Users', filters }));
+  }, [typeFilters, statusFilters]);
+  useEffect(() => {
+    dispatch(sortData({table: 'Users', column: sortBy }));
+  }, [sortBy]);
+  useEffect(() => {
+    const f = {
+      column: /^[0-9a-fA-F]{1,16}$/.test(search) ? 'id' : 'surname',
+      values: [ search ],
+    }
+    dispatch(filterData({table: 'Users', filters: [ f ]}));
+  }, [search]);
```

Comment:
if you will put one string in search and then remove it then the filter function returns empty results set - this is something probably to be slightly changed in the filterData function

easy workaround - just applying any filter brings the users back

Reply:
nevermind, I had some old version of the branch locally, after I went to the table slice I spotted it's already fixed :)